tx/textual/internal/cbor/cbor.go

```diff
@@ -0,0 +1,238 @@
+// Package cbor implements just enough of the CBOR (Concise Binary Object
+// Representation, RFC 8948) to deterministically encode simple data.
+package cbor
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"io"
+	"math"
+	"sort"
+)
+
+const (
+	major_uint        byte = 0
+	major_negint      byte = 1
+	major_byte_string byte = 2
+	major_text_string byte = 3
+	major_array       byte = 4
+	major_map         byte = 5
+	major_tagged      byte = 6
+	major_simple      byte = 7
+)
+
+func encode_first_byte(major byte, extra byte) byte {
```

Comment:
nit: Let's create a follow up PR to camelCase these functions (or maybe even the whole thing)